pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -36,16 +38,56 @@ const (
 	failedToReadUserApprovalPolicyReason = "FailedToReadUserApprovalPolicy"
 	unableToCreateMURReason              = "UnableToCreateMUR"
 	unableToDeleteMURReason              = "UnableToDeleteMUR"
+	userBanningReason                    = "Banning"
+	userBannedReason                     = "Banned"
 	userDeactivatingReason               = "Deactivating"
 	userDeactivatedReason                = "Deactivated"
 	invalidMURState                      = "InvalidMURState"
 	approvedAutomaticallyReason          = "ApprovedAutomatically"
 	approvedByAdminReason                = "ApprovedByAdmin"
 	pendingApprovalReason                = "PendingApproval"
+	failedToReadBannedUsersReason        = "FailedToReadBannedUsers"
+	missingUserEmailLabelReason          = "MissingUserEmailLabel"
+	missingEmailHashLabelReason          = "MissingEmailHashLabel"
+	invalidEmailHashLabelReason          = "InvalidEmailHashLabel"
 )
 
 var log = logf.Log.WithName("controller_usersignup")
 
+type BannedUserToUserSignupMapper struct {
+	client client.Client
+}
+
+func (b BannedUserToUserSignupMapper) Map(obj handler.MapObject) []reconcile.Request {
+	if bu, ok := obj.Object.(*toolchainv1alpha1.BannedUser); ok {
+		// look-up any associated UserSignup using the BannedUser's "toolchain.dev.openshift.com/emailHash" label
+		if emailHashLbl, exists := bu.Labels[toolchainv1alpha1.BannedUserEmailHashLabelKey]; exists {
+
+			labels := map[string]string{toolchainv1alpha1.UserSignupUserEmailHashAnnotationKey: emailHashLbl}
+			opts := client.MatchingLabels(labels)
+			userSignupList := &toolchainv1alpha1.UserSignupList{}
+			if err := b.client.List(context.TODO(), userSignupList, opts); err != nil {
+				log.Error(err, "Could not list UserSignup resources with label value", toolchainv1alpha1.UserSignupUserEmailHashAnnotationKey, emailHashLbl)
+				return nil
+			}
+
+			req := []reconcile.Request{}
+
+			for _, userSignup := range userSignupList.Items {
+				req = append(req, reconcile.Request{
+					NamespacedName: types.NamespacedName{Namespace: "toolchain-host-operator", Name: userSignup.Name},
+				})
+			}
+
+			return req
+		}
```

Comment:
I wonder if we it's actually OK if we just add the email-hash label here if it's not set... Maybe in a separate go routine..

Reply:
I actually do like this idea, and I'm sure it would help to reduce the difficulty in manual creation of UserSignup resources.  I'll spend a little time exploring this option to see if we break any rules by doing so.

Reply:
I've decided against doing this for now, I wasn't able to find any examples of this being done in any of my research, plus if we go with the CLI admin command for banning users it will be handled there.